assets/src/components/plugins-context-provider/use-normalized-plugins-data.js

```diff
@@ -0,0 +1,40 @@
+/**
+ * WordPress dependencies
+ */
+import { useContext, useEffect, useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { Plugins } from './index';
+
+export function useNormalizedPluginsData() {
+	const { fetchingPlugins, plugins } = useContext( Plugins );
+	const [ normalizedPluginsData, setNormalizedPluginsData ] = useState( [] );
+
+	useEffect( () => {
+		if ( fetchingPlugins || plugins.length === 0 ) {
+			return;
+		}
+
+		setNormalizedPluginsData( plugins.reduce( ( acc, source ) => {
+			const slug = source?.plugin?.match( /^(?:[^\/]*\/)?(.*?)$/ )?.[ 1 ];
```

Comment:
This doesn't seem quite right. If the plugin is `amp/amp.php`, will this not return `amp.php` instead of `amp`?

Reply:
Yes, you're right. I haven't considered such a pattern. How about this one:
```suggestion
			const slug = source?.plugin?.match( /^(?:[^\/]*\/)*([^.]*)/ )?.[ 1 ];
```
It should also take care of (a probably impossible) case when the PHP file is in a subdirectory like: `amp/src/amp.php`.

Reply:
I've actually refactored out the plugin slug retrieval to an external helper and used the regex as above in cfcbc92.

You'll also find there a minor update to the logic that removes Gutenberg and AMP from the plugins slugs. It's now done after parsing all the sources. This way even if Gutenberg is first in the array, it will be removed when there are more plugins. AMP is removed unconditionally.